test/unit/parameter_test.rb

```diff
@@ -44,6 +50,14 @@ class ParameterTest < ActiveSupport::TestCase
     host.clear_host_parameters_cache!
     assert_equal "dog", host.host_params["animal"]
 
+    subnet.subnet_parameters << SubnetParameter.create(:name => "animal", :value => "pigeon")
```

Comment:
subnet is nil here and that causes the tests to fail.
This can be fixed by adding `:with_subnet` to the host creation on line 30:
```ruby
host = FactoryGirl.create(:host, :with_hostgroup, :managed, :with_subnet)
```